sharding/indexHashedNodesCoordinatorWithRater.go

```diff
@@ -0,0 +1,142 @@
+package sharding
+
+import (
+	"fmt"
+	"github.com/ElrondNetwork/elrond-go/core"
+)
+
+type indexHashedNodesCoordinatorWithRater struct {
+	indexHashedNodesCoordinator
```

Comment:
seems we only require the fields in the indexHashedNodesCoordinator and not the implemented interfaces from it, then maybe you can just use the fields directly, and the constructor doesn't need to call new for the embedded struct.